src/postgraphile/http/subscriptions.ts

```diff
@@ -222,47 +215,39 @@ export async function enhanceHttpServerWithSubscriptions<
           // The original headers must win (for security)
           ...request.headers,
         };
+
+        return true;
       },
-      // tslint:disable-next-line no-any
-      async onOperation(message: any, params: ExecutionParams, socket: WebSocket) {
-        const opId = message.id;
-        const context = await getContext(socket, opId);
+      onSubscribe: async ({ socket }, message, args) => {
+        const context = await getContext(socket, message.id);
 
         // Override schema (for --watch)
-        params.schema = await getGraphQLSchema();
+        args.schema = await getGraphQLSchema();
 
-        Object.assign(params.context, context);
+        // if the context value is missing, initialise it
+        args.contextValue = { ...args.contextValue, ...(context as object) }; // mixed?
 
-        const { req, res } = await reqResFromSocket(socket);
         const meta = {};
-        const formatResponse = <TExecutionResult extends ExecutionResult = ExecutionResult>(
-          response: TExecutionResult,
-        ): TExecutionResult => {
+
+        const { req, res } = await reqResFromSocket(socket);
+        const executionResultFormatter: ExecutionResultFormatter = (_ctx, response) => {
           if (response.errors) {
             response.errors = handleErrors(response.errors, req, res);
           }
           if (!isEmpty(meta)) {
             response['meta'] = meta;
           }
-
           return response;
         };
-        // onOperation is only called once per params object, so there's no race condition here
-        // eslint-disable-next-line require-atomic-updates
-        params.formatResponse = formatResponse;
+
         const hookedParams = pluginHook
-          ? pluginHook('postgraphile:ws:onOperation', params, {
+          ? pluginHook('postgraphile:ws:onSubscribe', args, {
```

Comment:
Please don't rename the hook or the options unless there's definitely a breaking change here and the old one needs to be removed (in which case we need to have the old one throw when someone else hooks it) as this is a public API.

Reply:
I changed the name intentionally because the legacy `onOperation` func is different from the current `onSubscribe` func. Using the new lib is a breaking change anyway, so why not clean up?

Reply:
Okay. Be aware we definitely cannot merge this into v4 then; you might be better off doing it as a PR against the v5 branch (which is restructured as a monorepo).

Reply:
Its rebased on v4 because I am using this fork in production. I also see this breaking change better fitted for V5. I'll make a PR there too 👍 .

On the side note, how stable is V5? Would you recommend using it in production? Asking this because if I move the PR there too, would it make sense to test it out in production too?

Reply:
Don't try and use V5 yet ;)